app/main/views/agreements.py

```diff
@@ -62,28 +62,43 @@ def list_agreements(framework_slug):
 @login_required
 @role_required('admin', 'admin-ccs-sourcing')
 def next_agreement(supplier_id, framework_slug):
+    status = request.args.get("status")
```

Comment:
It's a necessary distinction, but perhaps not for the admin app. If countersigned agreements were instantly created and uploaded to S3 when Admin hits the button then there would be no need, but as it is the countersigned document is not returned to the supplier until some time after Admin hits the button, and so the supplier frontend needs some way to distinguish if it should show a link to download a countersigned document or not.